### ISSUE TITLE
Inlay hint request fails with "Debug Failure. Unexpected node"

### ISSUE THREAD

**Issue by yioneko** [yioneko/vtsls#122]
Example error log:
```
Error: Debug Failure. Unexpected node.
Node MethodSignature was unexpected.
     at visitForDisplayParts
     at Array.forEach (<anonymous>)
     ...
     at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
```

Upstream issue https://github.com/microsoft/TypeScript/issues/56276, introduced in TypeScript 5.3. The same error could also be reproduced in VSCode.

To circumvent this issue temporarily, you can downgrade vtsls to previous version, that the bundled version of TypeScript is still 5.2. Or using the workspace version of TypeScript instead of the bundled one by setting the following configuration:
```json
"typescript.tsdk": "node_modules/typescript/lib"
```

**Comment by jinzhongjia** [comment]
我看上游issue已经修复了该问题，是不是可以修复了

**Comment by yioneko** [comment]
@jinzhongjia 这个修复的话没有包含在最新的 release 里，可能要等到上游的 5.4 版本。

**Comment by jinzhongjia** [comment]
上游已经发布了新版

**Comment by yioneko** [comment]
Fixed in #140
